option unwrapping an interface results in compile error

### Describe the bug

Option unwrapping an interface results in a compile error.

### Reproduction Steps

```v
interface Interface {
	val int
}

struct Struct {
	val int
}

struct Foo {
	a ?Interface
	b ?Struct
}

fn main() {
	foo := Foo{
		a: Struct{val: 123}
		b: Struct{val: 456}
	}
	if foo.a != none {
		println(foo.a.val)
	}
	if foo.b != none {
		println(foo.b.val)
	}

}

```

### Expected Behavior

```
123
456
```

### Current Behavior

C Compile Error.

```
================== C compilation error (from gcc): ==============
cc: \Temp\v_0\proj.c: In function 'main__main':
cc: \Temp\v_0\proj.c:14100:54: error: implicit declaration of function 'I_main__Interface_as_I_main__Interface'; did you mean 'I_main__App_to_Interface_main__Interface'? [-Werror=implicit-function-declaration]
cc: 14100 |                 main__Interface b = (*(main__Interface*)(I_main__Interface_as_I_main__Interface(app->my_opt_interface));
cc:       |                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
```

### Possible Solution

Check ``!is_option_unwrap`` in the interface cast if-statement in cgen.v

https://github.com/pisaiah/v/tree/patch-1

### Additional Information/Context

_No response_

### V version

V 0.4.8

### Environment details (OS name and version, etc.)

Windows 11 x86-64
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4059,7 +4059,8 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 							g.write('*')
 						}
 						cast_sym := g.table.sym(g.unwrap_generic(typ))
-						if field_sym.kind == .interface && cast_sym.kind == .interface && !is_option_unwrap {
+						if field_sym.kind == .interface && cast_sym.kind == .interface
+							&& !is_option_unwrap {
 							ptr := '*'.repeat(field.typ.nr_muls())
 							dot := if node.expr_type.is_ptr() { '->' } else { '.' }
 							g.write('I_${field_sym.cname}_as_I_${cast_sym.cname}(${ptr}${node.expr}${dot}${node.field_name}))')

--- a/vlib/v/tests/options/option_selector_unwrap_types_test.v
+++ b/vlib/v/tests/options/option_selector_unwrap_types_test.v
@@ -1,5 +1,9 @@
 type SumType = int | string
 
+interface Interface {
+	a int
+}
+
 struct Struct {
 	a int
 }
@@ -9,6 +13,7 @@ struct Foo {
 	b ?string
 	c ?SumType
 	d ?Struct
+	e ?Interface
 }
 
 fn test_main() {
@@ -18,6 +23,9 @@ fn test_main() {
 		c: SumType(123)
 		d: Struct{
 			a: 123
+		}
+		e: Struct{
+			a: 456
 		}
 	}
 	if w.a != none {
@@ -41,6 +49,12 @@ fn test_main() {
 	if w.d != none {
 		dump(w.d)
 		assert w.d.a == 123
+	} else {
+		assert false
+	}
+	if w.e != none {
+		dump(w.e)
+		assert w.e.a == 456
 	} else {
 		assert false
 	}

